code/datums/components/anti_magic.dm

```diff
@@ -15,12 +15,43 @@
 	holy = _holy
 
 /datum/component/anti_magic/proc/on_equip(datum/source, mob/equipper, slot)
+	if(slot == SLOT_IN_BACKPACK)
+		return
```

Comment:
You need to unregister the signal rather than just returning since it could already be registered when moving to the backpack